axelor-account/src/main/java/com/axelor/apps/account/service/fixedasset/FixedAssetGenerationServiceImpl.java

```diff
@@ -272,11 +276,19 @@ public void generateAndComputeFixedAssetLines(FixedAsset fixedAsset) throws Axel
       } else {
         fixedAsset.setStatusSelect(FixedAssetRepository.STATUS_DRAFT);
       }
-      fixedAsset.setAcquisitionDate(invoice.getInvoiceDate());
+      fixedAsset.setQty(invoiceLine.getQty());
+      fixedAsset.setAcquisitionDate(invoice.getOriginDate());
       fixedAsset.setFirstDepreciationDate(invoice.getInvoiceDate());
       fixedAsset.setFirstServiceDate(invoice.getInvoiceDate());
       fixedAsset.setReference(invoice.getInvoiceId());
-      fixedAsset.setName(invoiceLine.getProductName() + " (" + invoiceLine.getQty() + ")");
+      if (invoiceLine.getQty() != null) {
+        fixedAsset.setName(
+            invoiceLine.getProductName()
+                + " ("
+                + invoiceLine.getQty().setScale(appBaseService.getNbDecimalDigitForQty())
```

Comment:
please add a rounding mode to manage the case where the config is changed to a scale lesser than the scale in database. Else in this case it will throw a runtime exception.

Reply:
done